packages/search-in-workspace/src/node/ripgrep-search-in-workspace-server.ts

```diff
@@ -93,47 +151,98 @@ export class RipgrepSearchInWorkspaceServer implements SearchInWorkspaceServer {
+                    globArguments.forEach(arg => args.add(arg));
+                }
+            });
+        };
+
+        args.add('--hidden');
+        args.add('--json');
+
+        args.add(options && options.matchCase ? '--case-sensitive' : '--ignore-case');
         if (options && options.includeIgnored) {
-            args.push('--no-ignore');
+            args.add('--no-ignore');
         }
         if (options && options.maxFileSize) {
-            args.push('--max-filesize=' + options.maxFileSize.trim());
+            args.add('--max-filesize=' + options.maxFileSize.trim());
         } else {
-            args.push('--max-filesize=20M');
+            args.add('--max-filesize=20M');
         }
+
         if (options && options.include) {
-            for (const include of options.include) {
-                if (include !== '') {
-                    args.push('--glob=**/' + include);
-                }
-            }
+            appendGlobArgs(options.include, false);
         }
+
         if (options && options.exclude) {
-            for (const exclude of options.exclude) {
-                if (exclude !== '') {
-                    args.push('--glob=!**/' + exclude);
-                }
-            }
+            appendGlobArgs(options.exclude, true);
         }
+
         if (options && options.useRegExp || options && options.matchWholeWord) {
-            args.push('--regexp');
+            args.add('--regexp');
         } else {
-            args.push('--fixed-strings');
-            args.push('--');
+            args.add('--fixed-strings');
+            args.add('--');
         }
-        return args;
+
+        return Array.from(args);
     }
 
-    // Search for the string WHAT in directories ROOTURIS.  Return the assigned search id.
+    /**
+     * Transforms a given file pattern to 'ripgrep' glob CLI arguments.
+     */
+    protected patternToGlobCLIArguments(pattern: string, exclude: boolean): string[] {
+        const toCLIGlobArgument = (inputPattern: string): string => {
+            const globCommandArgument = '--glob=';
+            const excludeChar = exclude ? '!' : '';
+            const subDirGlobPattern = '**/';
+
+            const subDirGlobPrefix = inputPattern.startsWith('/') ? '**' : subDirGlobPattern;
+            const updatedPattern = inputPattern.startsWith(subDirGlobPattern) ?
+                inputPattern : `${subDirGlobPrefix}${inputPattern}`;
+
+            return `${globCommandArgument}${excludeChar}${updatedPattern}`;
+        };
+
+        const globArgument = toCLIGlobArgument(pattern);
+
+        const globArgumentsArray = [globArgument];
+        if (!globArgument.endsWith('*')) {
+            // Add a generic glob CLI argument entry to include files inside a given directory.
+            const suffix = globArgument.endsWith('/') ? '*' : '/*';
+            globArgumentsArray.push(`${globArgument}${suffix}`);
```

Comment:
Can you explain why it's needed to push `globArgument` to `globArgumentsArray` here?

Reply:
A given pattern for include/ exclude files e.g. 'foo' can match a file or directory, the user may be explicit by adding a suffix e.g. '/*' to include files under a corresponding directory, but if the user does not specify the suffix, we add it here so we append two glob arguments e.g. 'foo' and 'foo/*'  this will match a file or files under the given directtory.